process/sync/baseForkDetector.go

```diff
@@ -470,3 +495,32 @@ func (bfd *baseForkDetector) shouldAddBlockInForkDetector(
 
 	return nil
 }
+
+func (bfd *baseForkDetector) computeForcedFork(
+	header data.HeaderHandler,
+	state process.BlockHeaderState) {
+	if state != process.BHProposed {
+		bfd.setForcedFork(false)
+		return
+	}
+
+	noncesDifference := int64(bfd.ProbableHighestNonce()) - int64(header.GetNonce())
+	isSyncing := noncesDifference > process.MaxNoncesDifference
+	if isSyncing {
+		bfd.setForcedFork(false)
```

Comment:
you do not need all these setForceFork to false. after it is set to TRUE, and forcesForkReset is processed, you set only there to FALSE.

Reply:
Yep, removed them